src/Rinds/RindsSpecsDataSourcePoller.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Handles polling and storage of RINDS specs
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\Rinds;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * RINDS specs data source poller class.
+ * This handles polling RINDS specs from JSON endpoints, and
+ * stores the specs in to the database as an option.
+ */
+class RindsSpecsDataSourcePoller {
+	const DATA_SOURCES = array(
+		'http://two.wordpress.test/rinds-specs.json',
+	);
+
+	/**
+	 * Polls the data sources for specs, persists those specs, and delete any
+	 * specs that no longer exist in the data sources.
+	 */
+	public static function poll_data_sources() {
+		$specs = array();
+
+		// Note that this merges the specs from the data sources based on the slug - last one wins.
+		foreach ( self::DATA_SOURCES as $url ) {
+			self::read_data_source( $url, $specs );
+		}
+
+		// Persist the specs as an option.
+		$existing_option = get_option( RindsEngine::SPECS_OPTION_NAME );
+
+		if ( false === $existing_option ) {
+			add_option( RindsEngine::SPECS_OPTION_NAME, $specs );
+		} else {
+			update_option( RindsEngine::SPECS_OPTION_NAME, $specs );
+		}
+
+		// Run the RINDS engine.
+		// @todo Break this out into it's own scheduled task.
+		RindsEngine::run();
+	}
+
+	/**
+	 * Read a single data source and insert the read specs into $specs, using
+	 * the spec slug as the key. So if the same or multiple data source has
+	 * a spec with the same slug, the last spec wins.
+	 *
+	 * @param string $url   The URL to read the specs from.
+	 * @param array  $specs The specs to read in to.
+	 */
+	private static function read_data_source( $url, &$specs ) {
+		$response = wp_remote_get( $url );
```

Comment:
This may be something for a follow-up, but I imagine we'll need to be able to pass headers or site information with this request.

Reply:
Do we have an idea of how many specs/notes will be stored in a source?

If the list is small, this may not be an issue.  But if data sources store many messages and aren't pruned over time, we'll also probably want to pass an id or a last fetched timestamp so we're not re-checking a huge data source over time.  Pagination could be also something to look at here.

Reply:
For a first version I wanted to keep this simple - just a static JSON file stored on a server. Making a more complex request and requiring a more complex server-side implementation seems like it should be a future piece of work.